cgi-bin/LJ/CleanHTML.pm

```diff
@@ -571,33 +573,47 @@ TOKEN:
                     return $text;
                 };
                 if ($cut) {
-                    my $etext = $link_text->();
-                    my $url   = LJ::ehtml($cut);
-                    $newdata .= "<div>" if $tag eq "div";
-
-                    # include empty span and div to be filled in on page
-                    # load if javascript is enabled
-                    $newdata .=
-                          "<span style=\"display: none;\" id=\"span-cuttag_"
-                        . $journal . "_"
-                        . $ditemid . "_"
-                        . $cutcount
-                        . "\" class=\"cuttag\"></span>";
-                    $newdata .= "<b>(&nbsp;<a href=\"$url#cutid$cutcount\">$etext</a>&nbsp;)</b>";
-                    $newdata .=
-                          "<div style=\"display: none;\" id=\"div-cuttag_"
-                        . $journal . "_"
-                        . $ditemid . "_"
-                        . $cutcount
-                        . "\" aria-live=\"assertive\">";
-                    $newdata .= "</div>";
-                    $newdata .= "</div>" if $tag eq "div";
-
-                    unless ( $opts->{'cutpreview'} ) {
-                        push @eatuntil, $tag;
+                    if ( $opts->{preserve_lj_tags} ) {
```

Comment:
Wait. I think I put this in the wrong branch of the enclosing "if". Now that I'm re-reading it, I think `$cut` means "this is displaying on last-N, so remove the cut content." That should never be enabled for crossposter.

Reply:
(In the process of fixing this, I rewrote things enough that I could just remove the second-pass parser in the crossposter itself.)